pkg/query-service/app/parser/metrics.go

```diff
@@ -36,6 +37,44 @@ func ParseMetricQueryRangeParams(r *http.Request) (*model.QueryRangeParamsV2, *m
 	if err := validateQueryRangeParamsV2(postData); err != nil {
 		return nil, &model.ApiError{Typ: model.ErrorBadData, Err: err}
 	}
+	// prepare the variables for the corrspnding query type
+	formattedVars := make(map[string]interface{})
+	for name, value := range postData.Variables {
+		if postData.CompositeMetricQuery.QueryType == model.PROM {
+			formattedVars[name] = metrics.PromFormattedValue(value)
+		} else if postData.CompositeMetricQuery.QueryType == model.CLICKHOUSE {
+			formattedVars[name] = metrics.FormattedValue(value)
```

Comment:
I see `PromFormattedValue` and `FormattedValue`  have the same input and output params which becomes an ideal candidate for extracting it as an interface and having 2 different implementations for that interface. The relevant interface will be called based on the type automatically without the `if` clause. Just a suggestion for observing patterns, might be an overkill now.

Reply:
Agree. I thought about it but went with this because same didn't apply for builder type.

Reply:
ok